src/main/flight/imu.c

```diff
@@ -342,3 +342,8 @@ int16_t calculateThrottleAngleCorrection(uint8_t throttle_correction_value)
         angle = 900;
     return lrintf(throttle_correction_value * sinf(angle / (900.0f * M_PIf / 2.0f)));
 }
+
+int16_t calculateThrottleAngleCorrectionAxis(uint8_t throttle_hovering_percent, uint8_t axis){
+    uint16_t liftOffTrust = throttle_hovering_percent * 10; //to bring from 0-100 range to 0-1000 range (thrust input is in the 1000-2000 range)
+    return lrintf( liftOffTrust*sinf(anglerad[axis]) );
```

Comment:
@bdoiron74 can you paste your code, so i can check your compensation value with mine? also cos() seems a bit complicated to use as it is 0 near +-90° and 1 at 0°, but maybe you are using a reference where 0° is flat, while for me 0° is engine pointing up.